text/0000-symbol-primitive.md

```diff
@@ -0,0 +1,120 @@
+- Feature Name: symbol_primitive
+- Start Date: 2020-12-08
+- RFC PR:
+- Neon Issue: [#502](https://github.com/neon-bindings/neon/issues/502)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an implementation for the `symbol` primitive: https://www.ecma-international.org/ecma-262/11.0/index.html#sec-ecmascript-language-types-symbol-type
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the `Symbol` function and primitive object wrapper API is available [using a workaround](https://github.com/neon-bindings/neon/issues/502#issuecomment-602568598) with `cx.global()`. However, typechecking a symbol passed in as an argument is not. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Constructing a `JsSymbol`:
+```rust
+// symbol with a description
+let symbol = JsSymbol::new(&mut cx, Some("foo"));
+
+// symbol without a description
+let symbol = JsSymbol::new(&mut cx, None::<&str>);
+
+// convenience construction method on Context
+let symbol = cx.symbol(Some("foo"));
+```
+
+As with other primitives, we can type check it:
+```rust
+fn is_symbol(mut cx: FunctionContext) -> JsResult<JsBoolean> {
+    let arg = cx.argument(0)?;
+    let result = arg.is_a::<JsSymbol,_>(&mut cx);
+    Ok(cx.boolean(result)) 
+}
+```
+
+Unlike many other primitives, `JsSymbol` doesn't have a `value` method to convert to a Rust type. However, it has a `description` method that returns the underlying `Symbol.prototype.description` instance property:
+```rust
+let symbol = JsSymbol::new(&mut cx, Some("foo"));
+let description = symbol.description(&mut cx);
+assert_eq!(description, Some("foo".to_owned()));
+
+let symbol_without_description = JsSymbol::new(&mut cx, None::<&str>);
+let description = symbol_without_description.description(&mut cx);
+assert_eq!(description, None);
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+#### JsSymbol
+- `new` would be implemented using [napi_create_symbol](https://nodejs.org/docs/latest-v14.x/api/n-api.html#n_api_napi_create_symbolhttps://nodejs.org/docs/latest-v14.x/api/n-api.html#n_api_napi_create_symbol) (as well as the v8 equivalent for the legacy-api)
```

Comment:
I think implementing `Symbol` for the legacy runtime should be considered _optional_ and left to the implementers discretion.